source/HailPanel.cpp

```diff
@@ -79,8 +79,8 @@ HailPanel::HailPanel(PlayerInfo &player, const shared_ptr<Ship> &ship, function<
 	{
 		// Is the player in any need of assistance?
 		const Ship *flagship = player.Flagship();
-		// Check if the player is out of fuel.
-		if(!flagship->JumpsRemaining())
+		// Check if the player is in need of fuel to jump away.
+		if(!flagship->JumpsRemaining() && flagship->JumpFuel() && flagship->Attributes().Get("fuel capacity"))
```

Comment:
Wouldn't we want to not just check that the ship has jump fuel and a fuel capacity but that the fuel capacity is enough for the jump fuel? Otherwise I think you could have a jump fuel of 200, a fuel capacity of 100, and a jumps remaining of 0, and you'd be offered to be refuel even though you can't be.
```suggestion
		// Check if the player is in need of fuel to jump away.
		double jumpFuel = flagship->JumpFuel();
		if(!flagship->JumpsRemaining() && jumpFuel && jumpFuel <= flagship->Attributes().Get("fuel capacity"))
```